erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -65,6 +66,38 @@ def on_cancel(self):
 		self.update_production_order()
 		self.make_gl_entries_on_cancel()
 
+	def validate_posting_date(self):
+		allow_negative_stock = cint(frappe.db.get_value("Stock Settings", None, "allow_negative_stock"))
+		items, warehouses = self.get_items_and_warehouses()
+		previous_stock_vouchers = self.get_previous_stock_vouchers(self.posting_date, self.posting_time, warehouses,
+		                                                           items)
+
+		if allow_negative_stock and previous_stock_vouchers:
+			for voucher_type, voucher_no in previous_stock_vouchers:
+				frappe.msgprint(
+					_("The posting date is after {0}: {1}, this will cause missing data in the General Ledger").format(
```

Comment:
the validation prevents stock entries being posted after there relevant sales invoice's so when the stock entry is updated the sales invoice's gl entries are also updated